packages/drivers/routerlicious-driver/src/documentStorageService.ts

```diff
@@ -183,3 +178,39 @@ export class DocumentStorageService implements IDocumentStorageService {
             : undefined;
     }
 }
+
+export class DocumentStorageService extends DocumentStorageServiceProxy {
```

Comment:
So if I got it right, it's DocumentStorageService  -> DocumentStorageServiceProxy  -> PrefetchDocumentStorageService -> DocumentStorageServiceCore (I'm mixing here inheritance and ownership, of course).

Should it be simply PrefetchDocumentStorageService  -> DocumentStorageService  (i.e. only change static creation method to conditionally wrap DocumentStorageService  , but leave structure as is otherwise the same)?

What are the pros and cons of both approaches?

Reply:
Yup, you've got it right. The second option you listed (PrefetchDocumentStorageService -> DocumentStorageService) was what I went for first, but then I realized that our DocumentDeltaStorageService utilizes the publicly exposed `logtailSha` property of DocumentStorageService. Wrapping DocumentStorageService in PrefetchDocumentStorageService hides this `logtailSha` property and breaks the DocumentDeltaStorageService, hence this more convoluted method of exposing logtailSha while still wrapping DocumentStorageServiceCore with PrefetchDocumentStorageService when enabled.

The other option I could see would be to not wrap `DocumentService`'s private `documentStorageService`, and only wrap what is returned by connectToStorage. That felt like a recipe for a hard-to-reproduce bug though

Reply:
I see. You could only wrap it when returning back storage to loader, that way internal connections stay the same. But that's worse in a sense that caching would happen only for calls coming from loader, vs. your solution makes it applicable to any calls to storage, including internal from driver, if any